File: Typescript5/1.cs
using System;

public class Program
{
    public static void Main(string[] args)
    {
        string a = "1011";
        string b = "110";
        Console.WriteLine("Binary sum of " + a + " and " + b + ": " + AddBinary(a, b));
    }

    public static string AddBinary(string a, string b)
    {
        string result = "";
        int i = a.Length - 1, j = b.Length - 1, carry = 0;

        while (i >= 0 || j >= 0 || carry > 0)
        {
            int digitA = i >= 0 ? a[i] - '0' : 0;
            int digitB = j >= 0 ? b[j] - '0' : 0;
            int sum = digitA + digitB + carry;
            result = (sum % 2) + result;
            carry = sum / 2;
            i--;
            j--;
        }

        return result;
    }
}

File: Typescript5/1.py

def ofcarry(a,b):

    result =""
    i,j,carry = len(a) -1, len(b)-1 , 0

    while i>=0 or j>=0 or carry >0:

        d1 = int(a[i]) if i>=0 else 0

        d2= int(b[j]) if j>=0 else 0

        sum = d1+d2+carry

        result = str(sum%2) + result

        carry = sum//2

        i -= 1
        j -= 1
    
    return result 

a="1011"
b="110"
print(ofcarry(a,b))



